Fix 195 Eclipse warnings about unnecessary code ... controversially!

liblit opened the pull request:
These changes fix a large number of Eclipse warnings about unnecessary code. However, unlike #156, the fixes here may be controversial. Perhaps we don't really want to fix them in the proposed manner, or even at all. For more details, please read the full description of each commit, not just each commit's one-line summary.

In general, if any of these proposed changes are undesirable, just let me know. I will revise the pull request to either suppress individual warnings or turn off the corresponding Eclipse diagnostic entirely.

liblit commented:
@msridhar, what specifically do you mean by “this one”? The entire pull request? One specific commit within the request? Certain specific changes within one of the rquest’s commits?

We can certainly suppress or disable anything you like, but I don’t want to assume that you’re asking for all 195 warnings to be suppressed if you really just mean a few of them or one warning category among the several categories addressed here.

msridhar commented:
Sorry I meant the one that gets rid of else blocks.

On Sat, Mar 25, 2017, 10:15 AM Ben Liblit <notifications@github.com> wrote:

> @msridhar <https://github.com/msridhar>, what specifically do you mean by
> “this one”? The entire pull request? One specific commit within the
> request? Certain specific changes within one of the rquest’s commits?
>
> We can certainly suppress or disable anything you like, but I don’t want
> to assume that you’re asking for all 195 warnings to be suppressed if you
> really just mean a few of them or one warning category among the several
> categories addressed here.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/wala/WALA/pull/157#issuecomment-289225672>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AALyUYxaugGxXrdGc6FLfGN4eDfLfjKpks5rpUuWgaJpZM4MoU6_>
> .
>

msridhar commented on com.ibm.wala.cast.java.ecj/src/com/ibm/wala/cast/java/translator/jdt/JDTJava2CAstTranslator.java:
```diff
@@ -287,16 +287,13 @@ public CAstEntity translateToCAst() {
 
-    private final Set<CAstAnnotation> annotations;
-    
     public ClassEntity(ITypeBinding jdtType, String name, Collection<CAstQualifier> quals, Collection<CAstEntity> entities,
```
We should also delete the unused constructor parameter and fix up callers.  Same for all changes in commit 934f8f524f228ecd5c8de7c6d9dac6a9efca4b7c where relevant

msridhar commented:
I added some more comments.  Also I'll probably cut release 1.4.1 before merging these changes as they affect some public APIs.

liblit commented:
@msridhar, thank you for taking my request for careful, skeptical review seriously. I will follow up with additional commits that address each item you flagged. Meanwhile, yes, I think it is perfectly sensible to cut a new release before merging these changes.

liblit commented:
@msridhar, I believe I have addressed all of your concerns as of commit 65be11f222672868300cdcf2d4fbd339c50f7a0a. You should wait for [the Travis check](https://travis-ci.org/wala/WALA/builds/215131704), of course. You may also want to look at 42c7866dfdfa152f174b8a8b462b3d290bfa195a with special care: while pruning unneeded code, it is possible that this commit either removed something with important side effects, or failed to remove something with no important side effects.

msridhar commented:
Awesome changes!  Merging.
